xds/src/main/java/io/grpc/xds/StructOrError.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2024 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.annotations.VisibleForTesting;
+import javax.annotation.Nullable;
+
+/** An object or a String error. */
+final class StructOrError<T> {
```

Comment:
Since we don't want new usages, why not mark it deprecated and include a comment that `StatusOr<T>` should be used instead?

Reply:
If I marked it `@Deprecated` then I'd need to fix all callers to at least suppress the deprecation. I don't think it is hurting much right now, but I didn't want new usages by non-gRPC XdsClient users. Then everything is in one repository and it is easier to clean up.